src/AlgebraicGeometry/ToricVarieties/NormalToricVarieties/standard_constructions.jl

```diff
@@ -410,25 +404,34 @@ Multivariate polynomial ring in 5 variables over QQ graded by
 ```
 """
 function blow_up(v::AbstractNormalToricVariety, n::Int; coordinate_name::String = "e", set_attributes::Bool = true)
-    new_fan = star_subdivision(polyhedral_fan(v), n)
-    new_variety = normal_toric_variety(new_fan)
-    new_rays = rays(new_fan)
-    old_rays = rays(v)
-    old_vars = string.(symbols(cox_ring(v)))
-    @req !(coordinate_name in old_vars) "The name for the blowup coordinate is already taken"
-    new_vars = Vector{String}(undef, length(new_rays))
-    for i in 1:length(new_rays)
-        j = findfirst(==(new_rays[i]), old_rays)
-        new_vars[i] = j !== nothing ? old_vars[j] : coordinate_name
-    end
-    if set_attributes
-      set_attribute!(new_variety, :coordinate_names, new_vars)
-    end
-    return new_variety
+  return _blow_up(v, star_subdivision(v, n); coordinate_name = coordinate_name, set_attributes = set_attributes)
 end
 
 
 
+function _blow_up(v::AbstractNormalToricVariety, new_fan::PolyhedralFan{QQFieldElem}; coordinate_name::String = "e", set_attributes::Bool = true)
+  new_variety = normal_toric_variety(new_fan)
+  new_rays = rays(new_fan)
+  old_rays = rays(v)
+  old_vars = string.(symbols(cox_ring(v)))
+  @req !(coordinate_name in old_vars) "The name for the blowup coordinate is already taken"
+  new_vars = Vector{String}(undef, length(new_rays))
+  for i in 1:length(new_rays)
+      j = findfirst(==(new_rays[i]), old_rays)
+      new_vars[i] = j !== nothing ? old_vars[j] : coordinate_name
+  end
+  if set_attributes
+    set_attribute!(new_variety, :coordinate_names, new_vars)
+  end
+  dim = ambient_dim(fan(v))
+  m = zero_matrix(ZZ, dim, dim)
+  for i in 1:dim
+    m[i, i] = one(ZZ)  
+  end
```

Comment:
Isn't there a predefined way to obtain the unit-matrix of size `dim` available?

Reply:
`identity_matrix(ZZ, dim)`, as defined in `AbstractAlgebra.jl`. Thank you for reminding me.